### DISCUSSION TITLE
Image progress in the watchface

### DISCUSSION THREAD

**Discussion by idimas13** [zepp-health/zeppos-forum#113] [Developer Feedback]
The watch doesn't show "image progress" pictures of the "Week PAI" module. 
![image](https://user-images.githubusercontent.com/54828913/224956952-a00e51e2-b040-4200-b5ce-5721dd0fad95.png)

**Comment by sunnyBob** [comment]
Please provide the device name and watch face resolution

**Reply by idimas13** [reply]
All ZeppOS devices, which I tested. For example, trex2, gtr4

**Reply by sunnyBob** [reply]
OK，Please help to confirm if the PAI value on the device is 0.

**Reply by idimas13** [reply]
This is soooo standard "support style" questions.... 
We are on github. Of course I checked everithing before post the problem.

**Reply by sunnyBob** [reply]
The reason for not displaying is related to the drawing logic of the current picture progress. It may be because your current pai value is too small. The maximum value of weekly pai is 75 (daily maximum) * 7 = 525. If the current number of pictures is 10, the first progress picture will be displayed at 5%-15%, which means at least 29 pai is required to display the first picture. You can refine the interval by increasing the number of pictures or using normal progress - color value progress.

**Reply by sunnyBob** [reply]
it is recommend adding a display effect on the background image for when the first progress picture is not yet displayed.

**Reply by idimas13** [reply]
Do you understand that this logic contradicts the meaning of the PAI indicator, for which 100 is a goal, and all the values ​​​​above are insignificant?

**Reply by idimas13** [reply]
I will try some math to match it...

**Reply by sunnyBob** [reply]
> Do you understand that this logic contradicts the meaning of the PAI indicator, for which 100 is a goal, and all the values ​​​​above are insignificant?

The current implementation is "current / max", and of course, I will provide feedback on your ideas to the firmware development team.

**Reply by idimas13** [reply]
I think in the watch face editor, the designer should be able to set his own logic of the image progress for most modules.
BUT, please, don't touch the heartrate)
